cmd/minishift/cmd/start.go

```diff
@@ -380,14 +372,56 @@ func clusterUp(config *cluster.MachineConfig) {
 		}
 	})
 
-	err = runner.Run(cmdName, cmdArgs...)
+	err := runner.Run(cmdName, cmdArgs...)
 	if err != nil {
 		// TODO glog is probably not right here. Need some sort of logging wrapper
 		glog.Errorln("Error starting the cluster: ", err)
 		atexit.Exit(1)
 	}
 }
 
+// Check for existing cache for oc binary and return oc binary path
+func ocCommand(config *cluster.MachineConfig) string {
+	oc := cache.Oc{
+		OpenShiftVersion:  config.OpenShiftVersion,
+		MinishiftCacheDir: filepath.Join(constants.Minipath, "cache"),
+	}
+	err := oc.EnsureIsCached()
+	if err != nil {
+		glog.Errorln("Error starting the cluster: ", err)
+		atexit.Exit(1)
+	}
+
+	cmdName := filepath.Join(oc.GetCacheFilepath(), constants.OC_BINARY_NAME)
+	return cmdName
+}
+
+// set Minishift current-context as "minishift"
+// https://docs.openshift.com/enterprise/3.0/cli_reference/manage_cli_profiles.html (set-context section)
+func setMinishiftContext(config *cluster.MachineConfig, host *host.Host) {
+	cmdName := ocCommand(config)
+	ip, _ := host.Driver.GetIP()
```

Comment:
@budhrg  In this stage VM is already up and running and surely have IP (which used for provision)  if we required it in before provision step then I agree with you to handle this error.